From 41f8e16425f65d07b777172844a7daac7a9d7f2f Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Thu, 20 Sep 2018 15:19:57 -0700
Subject: [PATCH 17/17] lineage: Move some kernel definitions to
 BoardConfigKernel

Kernel source settings should always come at BoardConfig population
time so things that use the variable later don't end up pointing to
an empty or wrong variable.

Change-Id: If046b86ff0d18c76898e90295be873a8379f678a
---
 build/tasks/kernel.mk        |  9 ---------
 config/BoardConfigKernel.mk  | 25 +++++++++++++++++++++++++
 config/BoardConfigLineage.mk |  2 ++
 3 files changed, 27 insertions(+), 9 deletions(-)
 create mode 100644 config/BoardConfigKernel.mk

diff --git a/build/tasks/kernel.mk b/build/tasks/kernel.mk
index 078ee1f0..3b2d0d1a 100644
--- a/build/tasks/kernel.mk
+++ b/build/tasks/kernel.mk
@@ -72,11 +72,7 @@
 
 ifneq ($(TARGET_NO_KERNEL),true)
 
-TARGET_AUTO_KDIR := $(shell echo $(TARGET_DEVICE_DIR) | sed -e 's/^device/kernel/g')
-
 ## Externally influenced variables
-# kernel location - optional, defaults to kernel/<vendor>/<device>
-TARGET_KERNEL_SOURCE ?= $(TARGET_AUTO_KDIR)
 KERNEL_SRC := $(TARGET_KERNEL_SOURCE)
 # kernel configuration - mandatory
 KERNEL_DEFCONFIG := $(TARGET_KERNEL_CONFIG)
@@ -87,12 +83,7 @@ SELINUX_DEFCONFIG := $(TARGET_KERNEL_SELINUX_CONFIG)
 KERNEL_OUT := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ
 KERNEL_CONFIG := $(KERNEL_OUT)/.config
 
-TARGET_KERNEL_ARCH := $(strip $(TARGET_KERNEL_ARCH))
-ifeq ($(TARGET_KERNEL_ARCH),)
-KERNEL_ARCH := $(TARGET_ARCH)
-else
 KERNEL_ARCH := $(TARGET_KERNEL_ARCH)
-endif
 
 ifeq ($(KERNEL_ARCH),x86_64)
 KERNEL_DEFCONFIG_ARCH := x86
diff --git a/config/BoardConfigKernel.mk b/config/BoardConfigKernel.mk
new file mode 100644
index 00000000..79dd6788
--- /dev/null
+++ b/config/BoardConfigKernel.mk
@@ -0,0 +1,25 @@
+# Copyright (C) 2018 The LineageOS Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+TARGET_AUTO_KDIR := $(shell echo $(TARGET_DEVICE_DIR) | sed -e 's/^device/kernel/g')
+
+# kernel location - optional, defaults to kernel/<vendor>/<device> (see above TARGET_AUTO_KDIR)
+TARGET_KERNEL_SOURCE ?= $(TARGET_AUTO_KDIR)
+
+CUSTOM_KERNEL_ARCH := $(strip $(TARGET_KERNEL_ARCH))
+ifeq ($(CUSTOM_KERNEL_ARCH),)
+TARGET_KERNEL_ARCH := $(TARGET_ARCH)
+else
+TARGET_KERNEL_ARCH := $(CUSTOM_KERNEL_ARCH)
+endif
diff --git a/config/BoardConfigLineage.mk b/config/BoardConfigLineage.mk
index fc509f29..547792fd 100644
--- a/config/BoardConfigLineage.mk
+++ b/config/BoardConfigLineage.mk
@@ -3,6 +3,8 @@ ifeq ($(WITH_LINEAGE_CHARGER),true)
     BOARD_HAL_STATIC_LIBRARIES := libhealthd.lineage
 endif
 
+include vendor/lineage/config/BoardConfigKernel.mk
+
 ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
 include vendor/lineage/config/BoardConfigQcom.mk
 endif
-- 
2.17.1

